core_gem/lib/deep_cover/core_ext/exec_callbacks.rb

```diff
@@ -1,14 +1,15 @@
 # frozen_string_literal: true
 
-# This file is required by absolute path in the entry_points when doing clone mode.
+# This file is used by projects cloned with clone mode. As such, special care must be taken to
+# be compatible with any projects.
 # THERE MUST NOT BE ANY USE/REQUIRE OF DEPENDENCIES OF DeepCover HERE
-# See deep-cover/core_gem/lib/deep_cover/setup/clone_mode_entry.rb for details
-
-require_relative '../module_override'
+# See deep-cover/core_gem/lib/deep_cover/setup/clone_mode_entry_template.rb for explanation of
+# clone mode and of this top_level_module stuff.
+top_level_module = Thread.current['_deep_cover_top_level_module'] || Object
 
 # Adds a functionality to add callbacks before an `exec`
 
-module DeepCover
+module top_level_module::DeepCover # rubocop:disable Naming/ClassAndModuleCamelCase
```

Comment:
:-1: Can't we use the instrumented code?

Reply:
I find it a nice goal that we could use any version of DeepCover to cover any other version of DeepCover. If we manage that, then I have more confidence in the general quality of the system.

We may never change this file, so it's maybe not necessary here. But why not do it here too? I need to do that for other parts of DeepCover, such as for Persistence and GlobalVariables, so one or two more file aren't hurting.